src/hooks/posters.ts

```diff
@@ -71,24 +72,55 @@ export const usePosters = (posterHallId: string) => {
   const fuseVenues = useMemo(
     () =>
       new Fuse(filteredPosterVenues, {
-        keys: ["poster.title", "poster.authorName", "poster.categories"],
+        keys: [
+          "name",
+          "poster.title",
+          "poster.authorName",
+          "poster.categories",
+        ],
+        threshold: 0.2, // 0.1 seems to be exact, default 0.6: brings too distant if anyhow related hits
+        ignoreLocation: true, // default False: True - to search ignoring location of the words.
+        findAllMatches: true,
       }),
     [filteredPosterVenues]
   );
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery)
-      return filteredPosterVenues.slice(0, displayedPostersCount);
+    const normalizedSearchQuery = searchQuery.trim();
+
+    if (!normalizedSearchQuery) return filteredPosterVenues;
+
+    const matchEntries = breakStringWithQuotesBySpaces(normalizedSearchQuery);
+
+    if (!matchEntries) return filteredPosterVenues;
 
     return fuseVenues
-      .search(searchQuery)
-      .slice(0, displayedPostersCount)
+      .search({
+        $and: matchEntries.map((x) => ({
```

Comment:
@yarikoptic @mike-lvov Instead of hacking our own workaround to match quoted strings together, why not enable/use the built in syntax/mode supported by Fuse.js?

- https://fusejs.io/examples.html#extended-search
  - > White space acts as an AND operator, while a single pipe (`|`) character acts as an OR operator. To escape white space, use double quote ex. `="scheme language"` for exact match.

Reply:
Tried playing around with this a little (with extended search enabled) using the search term `eeg`.

---

Just using:

```typescript
.search(normalizedSearchQuery)
```

---

Just using a top level `$or` for the fields, with no 'custom tokenisation' fails when trying to use a search term and the authors name to narrow it down (eg. `eeg rux` fails to find the poster I want)

Presumably because of it's 'short circuit':

- https://fusejs.io/api/query.html#or
  - > The `$or` operator performs a logical OR operation on an array expressions and selects the entries that satisfy at least one of the expressions. **The `$or` operator uses short-circuit evaluation (i.e, if the first expression evaluates to true, Fuse.js will not evaluate the remaining expressions).**

```typescript
.search({
  $or: [
    { name: normalizedSearchQuery },
    { "poster.title": normalizedSearchQuery },
    { "poster.authorName": normalizedSearchQuery },
    { "poster.categories": normalizedSearchQuery },
  ],
})
```

---

I think I played around with some other variations of things, basically attempting to find a method of using the extended search features without needing to tokenise the search string ourselves.. but.. basically didn't manage to figure anything useful/it all seemed to work less good than the 'custom tokenisation' version (even though it feels weirdly hacky that we would need to do that for some reason.. maybe just because the lib isn't so much designed for passing user-facing search input directly through)